bert/src/tokenizer/roberta.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2023 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    fs::{read_to_string, File},
+    io::BufReader,
+};
+
+use ndarray::Array2;
+use tokenizers::{
+    models::unigram::Unigram,
+    normalizers::Precompiled,
+    pre_tokenizers::{
+        metaspace::Metaspace,
+        sequence::Sequence,
+        whitespace::WhitespaceSplit,
+        PreTokenizerWrapper,
+    },
+    processors::template::{TemplateProcessing, TemplateProcessingBuilder},
+    utils::{
+        padding::{PaddingDirection, PaddingParams, PaddingStrategy},
+        truncation::{TruncationDirection, TruncationParams, TruncationStrategy},
+    },
+    Error,
+    Model,
+    TokenizerBuilder,
+    TokenizerImpl,
+};
+
+use crate::{
+    config::Config,
+    tokenizer::{Encoding, Tokenize},
+};
+
+/// A pre-configured Roberta tokenizer.
+pub struct Tokenizer(TokenizerImpl<Unigram, Precompiled, Sequence, TemplateProcessing, Metaspace>);
+
+impl Tokenize for Tokenizer {
+    fn new<P>(config: &Config<Self, P>) -> Result<Self, Error> {
+        let vocab = serde_json::from_reader::<_, Vec<(String, f64)>>(BufReader::new(File::open(
+            config.dir.join("vocab.txt"),
+        )?))?;
+        let unk_token = config.extract::<String>("tokenizer.tokens.unknown")?;
+        let unk_id = vocab.iter().position(|(word, _)| word == &unk_token);
+        let model = Unigram::from(vocab, unk_id)?;
```

Comment:
:art: 

```suggestion
        let unknown_token = config.extract::<String>("tokenizer.tokens.unknown")?;
        let unknown_id = vocab.iter().position(|(word, _)| word == &unknown_token);
        let model = Unigram::from(vocab, unknown_id)?;
```

Reply:
the naming has been like this for the other tokenizer as well, but writing it out is ok as well, i'll change it at all places.